src/SharedResources.cpp

```diff
@@ -24,6 +24,7 @@ static bool SharedResourceExecute(const Frozen::SharedResourceData *sharedResour
     uint64_t time_exec_started = TimerGet();
     ExecResult result = ExecuteProcess(action, envVarsCount, envVars, heap, 0, false);
     PrintMessage(result.m_ReturnCode == 0 ? MessageStatusLevel::Success : MessageStatusLevel::Failure, (int)TimerDiffSeconds(time_exec_started, TimerGet()), &result, fullAnnotation);
+    ExecResultFreeMemory(&result);
     return result.m_ReturnCode == 0;
```

Comment:
Probably OK, but it feels wrong to free memory and then read from it.

Reply:
No, this is fine. `ExecResultFreeMemory` frees the memory allocated by `ExecResult` (the output buffer), but the `ExecResult` itself is a local struct, so `m_ReturnCode` is still valid.